code/modules/projectiles/projectile.dm

```diff
@@ -792,6 +789,11 @@ So if we are on the 32th absolute pixel coordinate we are on tile 1, but if we a
 			if(living_soft_armor)
 				living_soft_armor = max(0, living_soft_armor - penetration) //Flat removal.
 
+		var/mob/living/carbon/shooter_carbon = proj.firer
+		if(iscarbon(shooter_carbon))
```

Comment:
typecasting before checking could lead to runtimes if I recall correctly.

Reply:
Should be fine; it would get caught in the check, but I don't mind changing this.